tests/hawkey/test_sack.cpp

```diff
@@ -83,13 +83,13 @@ START_TEST(test_list_arches)
     const char ** arches = dnf_sack_list_arches(sack);
 
     /* noarch, x86_64, athlon, i686, i586, i486, i386 */
-    fail_unless(g_strv_length((gchar**)arches) == 7);
+    fail_unless(7 - g_strv_length((gchar**)arches) >= 0);
```

Comment:
Hello, first, why do you make the check unnecessarily less readable? It's equal to `g_strv_length((gchar**)arches) <= 7`.

Second, this will now pass even if the arch list is empty, so this should at least check le length to be 6 or 7.

But it would be better if it actually checked the contents of the list and checked the presence of the 6 stable arches, plus, if there's the seventh one, check that it's "athlon".

Reply:
@lukash hello, what about this version? I'm using a lower limit of 6 for the check on number of arches and I'm checking if "athlon" value if provided in case of seven arches.